ocaml/libs/rate-limit/bucket_table.ml

```diff
@@ -0,0 +1,158 @@
+(*
+ * Copyright (C) 2025 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+type rate_limit_data = {
+    bucket: Token_bucket.t
+  ; process_queue:
```

Comment:
Would it be possible to let the caller know how long it should delay before moving ahead?
For example:
```
consume t amount |> Option.iter Thread.delay ;
handle ()
```
The schedules of different threads can be partially sorted by the delays returned to them.